pkg/operator/ceph/csi/spec.go

```diff
@@ -586,9 +597,20 @@ func createCSIDriverInfo(ctx context.Context, clientset kubernetes.Interface, na
 			PodInfoOnMount: &mountInfo,
 		},
 	}
+	if fsGroupPolicy != "" {
+		policy := k8scsi.FSGroupPolicy(fsGroupPolicy)
+		csiDriver.Spec.FSGroupPolicy = &policy
+	}
 	csidrivers := clientset.StorageV1beta1().CSIDrivers()
 	driver, err := csidrivers.Get(ctx, name, metav1.GetOptions{})
 	if err == nil {
+		// As FSGroupPolicy field is immutable, should be set only during
+		// create time.
+		// if the request is to change the FSGroupPolicy, we are deleting the CSIDriver object and creating it.
+		if driver.Spec.FSGroupPolicy != nil && csiDriver.Spec.FSGroupPolicy != nil && *driver.Spec.FSGroupPolicy != *csiDriver.Spec.FSGroupPolicy {
```

Comment:
If they are enabling the setting for the first time (changing the policy from blank to some value), or disabling the setting (changing the policy back to blank), will the driver be recreated? It seems like it will only be updated if the value is already set and is being changed.

Reply:
It supports only 3 values as per the doc I linked above, setting empty value is not allowed. If the kubernetes is upgraded from 1.18 to 1.19 the csidriver value will be set to default as per my testing (I can test the upgrade again and see)

Reply:
On line 600 above it looks like the policy could be blank. That would at least happen on upgrades, right? So if they upgraded and then applied the new policy, the csi driver wouldn't be restarted?

Reply:
After upgrade kubernetes will set the value to ReadWriteOnceWithFSType which is default. The check is for safer side and for kubernetes versions less than 1.19, even if we set the fsgrouppolicy it will be discarded.

Reply:
User is not suppose to set it to empty in configmap ,if he set it to empty the object update will fail. Its admin responsibility to set it to valid options.

Reply:
Where will it fail if the value is not set in the configmap? It seems like it just won't update the driver in that case. 

The Rook upgrade does not require adding the values to the configmap so the settings may not be in the configmap at all. 

The default values (the last param) to load the settings on lines 490 and 501 are blank. Do we just need to set those to the expected default value? Then we will guarantee it's always set.

Reply:
added the default value now and also logging what Rook is defaulting.